### DISCUSSION TITLE
Creating a extension for showing comment outlines

### DISCUSSION THREAD

**Discussion by AnasGrzor** [microsoft/vscode-discussions#1194] [Extension Development QnA] [answered]
I'm trying to create a extension which creates a outline around comments. I'm having trouble with multiline comments. I want to enclose the multiline comment in a single box. Currently it's creating a border at every line. I'm building on top of the better comments extension. 
![image](https://github.com/microsoft/vscode-discussions/assets/103470289/2ed48848-887a-4f60-83c4-326eceaf1ec3)
This is how it should look like
![image](https://github.com/microsoft/vscode-discussions/assets/103470289/f0c94173-e986-46f7-a134-559f1d93b003)

If anyone can help me, I would appreciate it.

**Comment by alefragnani** [comment, marked as answer]
Hi @AnasGrzor ,

As far as I remember, decorators are applied _per line_. This means, in order to have a _box like_ representation, it would be required to create 3 different decorators:

* one to the top line, with borders at top, left and right 
* one to the middle lines, with borders at left and right
* one to the bottom line, with borders at bottom, left and right

> This issue from 2019 (https://github.com/microsoft/vscode/issues/78640) has a similar request and the direction was to split decorators

Even tho, I'm not sure it is possible to have the right borders, in the multiline comment, vertically aligned. Decorators tends to go only until the last character in the line, or the `isWholeLline`, but not a specific width. I have played with it a bit in the past on both of my extensions (Bookmarks and Separators), with no success. If you are able to find a way please, share it, if possible.

Hope this helps

**Reply by AnasGrzor** [reply]
VS code does not offer us the option to set borders at the top, left, bottom, right. I did try to create something using the before and after pseudoselectors but it failed horribly. So there's no hope for this. I'll tell my client that and hope that's it. I'll mark this as an answer and close this.

**Reply by alefragnani** [reply]
It does supports top and bottom. You can look at my Separators extension (https://marketplace.visualstudio.com/items?itemName=alefragnani.separators), as I use both.

I'm not in front of my computer to try out combining left and right borders, but you may try looking the `borderWidth` attribute. 

I remember an extension that was trying to create a box around brackets, but using colored background, not borders. I don't remember its name, neither if it was successful in its attempt, unfortunately.

**Comment by abrilurena** [comment]
Hi, @AnasGrzor!  do you still need help with this?
